crates/parser/src/lib.rs

```diff
@@ -465,6 +465,27 @@ impl Interface {
             _ => false,
         }
     }
+    pub fn get_variant(&self, ty: &Type) -> Option<&Variant> {
+        if let Type::Id(id) = ty {
+            match &self.types[*id].kind {
+                TypeDefKind::Variant(v) => Some(v),
+                _ => None,
+            }
+        } else {
+            None
+        }
+    }
+
+    pub fn get_nullable_option(&self, ty: &Type) -> Option<&Type> {
+        self.get_variant(ty).and_then(|v| v.as_option())
+    }
+
+    pub fn is_nullable_option(&self, variant: &Variant) -> bool {
+        variant.as_option().map_or(false, |ty| {
+            self.get_nullable_option(ty)
+                .map_or(true, |ty| self.get_nullable_option(ty).is_none())
+        })
+    }
```

Comment:
I don't think that this should live in the `wit-parser` crate because this is a JS-specific operation. Options are represented differently in each language and each one may need its own special casing.

Reply:
While I agree that different languages will represent them differently, this allows any language to check if the type is `option<T>` vs `option<option<T>>`, which isn't about representation but rather a fact about the variant.

Reply:
Perhaps, yeah, but I would personally prefer that this remain language-specific for now and in the future if there's duplication between languages we can deduplicate at that time.